services/scan/src/store.ts

```diff
@@ -375,6 +381,95 @@ export class Store {
     );
   }
 
+  setBatchesAsBackedUp() {
+    this.setBackupTimestamp(BackupKey.Batches);
+  }
+
+  setCvrsAsBackedUp() {
+    this.setBackupTimestamp(BackupKey.Cvrs);
+  }
+
+  /**
+   * Sets the timestamp of a backup
+   */
+  private setBackupTimestamp(key: BackupKey): void {
+    this.client.run(
+      'insert or replace into backups (key, value) values (?, current_timestamp)',
+      key
+    );
+  }
+
+  /**
+   * Gets a config value by key.
+   */
+  private getBackupTimestamp(key: BackupKey): Iso8601Timestamp | undefined {
+    const row = this.client.one<[Iso8601Timestamp]>(
+      'select value from backups where key = ?',
+      key
+    ) as { value: Iso8601Timestamp } | undefined;
+    return row?.value;
+  }
+
+  /**
+   * Returns whether the appropriate backups have been completed and it is safe
+   * to unconfigure a machine/zero out data. Always returns true in test mode.
+   */
+  getCanUnconfigure(): boolean {
+    // Always allow unconfiguring a machine in test mode
+    if (this.getTestMode()) {
+      return true;
+    }
+
+    const batchesBackedUpAt = this.getBackupTimestamp(BackupKey.Batches);
+    const cvrsBackedUpAt = this.getBackupTimestamp(BackupKey.Cvrs);
+
+    if (!this.batchStatus().length) {
+      return true;
+    }
+
+    // Require both batches and CVRs to have been backed up
+    if (!(cvrsBackedUpAt && batchesBackedUpAt)) {
+      return false;
+    }
+
+    const { maxCvrsCreatedAt } = this.client.one(
+      'select max(created_at) as maxCvrsCreatedAt from sheets'
+    ) as { maxCvrsCreatedAt: Iso8601Timestamp };
+    const { maxCvrsDeletedAt } = this.client.one(
+      'select max(deleted_at) as maxCvrsDeletedAt from sheets'
+    ) as { maxCvrsDeletedAt: Iso8601Timestamp };
+    const { maxBatchesStartedAt } = this.client.one(
+      'select max(started_at) as maxBatchesStartedAt from batches'
+    ) as { maxBatchesStartedAt: Iso8601Timestamp };
+    const { maxBatchesDeletedAt } = this.client.one(
+      'select max(deleted_at) as maxBatchesDeletedAt from batches'
+    ) as { maxBatchesDeletedAt: Iso8601Timestamp };
```

Comment:
You could do this in two SQL statements, e.g.:

`select max(created_at) as maxCVrsCreatedAt, max(deleted_at) as maxCvrsDeletedAt from sheets`

Reply:
Good idea! Done.